tests/common/config_reload.py

```diff
@@ -77,4 +80,15 @@ def config_reload(duthost, config_source='config_db', wait=120, start_bgp=True,
     modular_chassis = duthost.get_facts().get("modular_chassis")
     wait = max(wait, 240) if modular_chassis else wait
 
-    time.sleep(wait)
+    if safe_reload:
+        # The wait time passed in might not be guaranteed to cover the actual
+        # time it takes for containers to come back up. Therefore, add 5
+        # minutes to the maximum wait time. If it's ready sooner, then the
+        # function will return sooner.
+        pytest_assert(wait_until(wait + 300, 20, 0, duthost.critical_services_fully_started),
+                "All critical services should be fully started!")
+        if config_source == 'minigraph':
```

Comment:
Is it possible to also add bgp peer states established check here as well using duthost.bgp_facts. This will ascertain that the ports are also initialized and operational.

Reply:
I think at this time, it might be better to have it be checked by the individual test cases if they want to make sure that BGP sessions are up, in case some services expect some BGP sessions to be down, but still want to do a safe reload..